### PR TITLE
RA2 - Added IFV

### PR DESCRIPTION
**Pull request by MustaphaTR**
As always made on tablet. ~~Needs  squash and test first.~~

Unable to change its turret yet. Also I'm not sure is it possible.

Tested, working well.

### CODE REVIEW

File: weapons.yaml [hunk 1]
```diff
@@ -1189,4 +1189,98 @@ Medusa:
...
+		Speed: 213
+		Arm: 2
+		Blockable: false
+		Shadow: true
+		Inaccuracy: 128
+		Image: DRAGON
+		Trail: small_smoke_trail
+		RateOfTurn: 8
+		RangeLimit: 35
+		Palette: ra
```
Comment (tomsons26) [inline weapons.yaml]: Don't add palette ra. The trailers and projectiles mostly use anim pal which is already the default for such effects

File: rules/allied-vehicles.yaml [hunk 2]
```diff
@@ -152,4 +152,52 @@ tnkd:
...
\ No newline at end of file
+	WithVoxelBody:
+
+ifv:
+	Inherits: ^Vehicle
+	Valued:
+		Cost: 600
+	Tooltip:
+		Name: Infantry Fighting Vehicle
+		Description: Allied Multi-Purpose Vehicle.\nStrong vs. Infantry, Aircraft\nWeak vs. Vehicles
```
Comment (Phrohdoh) [inline rules/allied-vehicles.yaml]: We can't really give a Strong/Weak vs as it depends entirely on the passenger.

File: rules/allied-vehicles.yaml [hunk 3]
```diff
@@ -152,4 +152,52 @@ tnkd:
...
+	Armament@ELITE:
+		Weapon: HoverMissileE
+		LocalOffset: 640,0,384
+		UpgradeTypes: eliteweapon
+		UpgradeMinEnabledLevel: 1
+	AttackTurreted:
+	AutoTarget:
+	Cargo:
+		Types: Infantry
+		MaxWeight: 20
```
Comment (Phrohdoh) [inline rules/allied-vehicles.yaml]: 1

**Comment by Phrohdoh** [comment]
These changes will now need to be made against the new layout.
In other words add this to `ra2/rules/allied-vehicles.yaml` and `ra2/weapons.yaml`, please.

**Comment by Phrohdoh** [comment]
Made these changes locally, thanks!

**Pull request closed without merging**